### PR TITLE
A couple of fixups for add-i-in-c-status-and-help

### PR DESCRIPTION
**Pull request by dscho**
Before integrating the built-in `add -p` into Git for Windows' `master`, I had actually forgotten about @jeffhostetler's very helpful suggestions. Better late than never ;-)

### CODE REVIEW

**Comment by jeffhostetler** [comment]
WRT add-interactive.c, I think this helps greatly.  Thanks!

File: prefix-map.c [hunk 1]
```diff
@@ -35,40 +35,40 @@ static void init_prefix_map(struct prefix_map *prefix_map,
...
 			    struct prefix_item *item)
 {
-	struct prefix_map_entry *e = xmalloc(sizeof(*e)), *e2;
+	struct prefix_map_entry e = { { NULL } }, *e2;
 	int j;
 
-	e->item = item;
-	e->name = e->item->name;
+	e.item = item;
+	e.name = item->name;
```
Comment (jeffhostetler) [inline prefix-map.c]: Maybe add:

If (strlen(e.name) < prefix_map->min_length) BUG(...);

Comment (jeffhostetler) [reply prefix-map.c]: (I’m viewing this on my iPad so I can’t tell if this case is handled elsewhere.)

File: prefix-map.c [hunk 2]
```diff
@@ -35,40 +35,40 @@ static void init_prefix_map(struct prefix_map *prefix_map,
...
 
-	for (j = prefix_map->min_length; j <= prefix_map->max_length; j++) {
-		if (!isascii(e->name[j])) {
-			free(e);
+	for (j = prefix_map->min_length;
+	     j <= prefix_map->max_length && e.name[j]; j++) {
+		/* Avoid breaking UTF-8 multi-byte sequences */
+		if (!isascii(e.name[j]))
```
Comment (jeffhostetler) [inline prefix-map.c]: Does this mean that we never put names with utf8 chars into the map?  The way I read this is that if there is a utf8 char anywhere between min and max, we skip the name, right?

**Comment by jeffhostetler** [comment]
WRT prefix-map.c, I studied it as best I could on my iPad. I think I need to wait until I get back to my computer to look at it in more detail.  But don’t let this block you.

File: prefix-map.c (hunk 1 above)
Comment (dscho) [reply prefix-map.c]: It is not, but I do not want to overdo the checks for invariants...

File: prefix-map.c (hunk 2 above)
Comment (dscho) [reply prefix-map.c]: We totally do.

**Pull request merged**